rest_framework/serializers.py

```diff
@@ -1627,6 +1633,29 @@ def get_unique_for_date_validators(self):
 
         return validators
 
+    def get_partial_update_extra_fields(self):
+        partial_update_extra_fields = getattr(self.Meta, 'partial_update_extra_fields', None)
+
+        if partial_update_extra_fields is not None and not isinstance(partial_update_extra_fields, (list, tuple)):
+            raise TypeError(
+                'The `partial_update_extra_fields` option must be a NoneType or list or tuple. Got %s.' %
+                type(partial_update_extra_fields).__name__
+            )
+
+        fields = self.get_fields()
+        if partial_update_extra_fields is not None:
+            for field_name in partial_update_extra_fields:
+                assert field_name in fields, (
+                    "The field '{field_name}' was included on serializer "
+                    "{serializer_class} in the 'partial_update_extra_fields' option, but does "
+                    "not match any model field.".format(
+                        field_name=field_name,
+                        serializer_class=self.__class__.__name__
+                    )
+                )
```

Comment:
Here it should use model's fields instead of those from serializer, or it would not make much sense to build any kind of "extra-update-fields" list.

There are already some utility methods in DRF, like this one in the serializers package.

https://github.com/encode/django-rest-framework/blob/33b86a8e5383ae7c6bba03cf09b12b8f7cadd20f/rest_framework/serializers.py#L1051)